base/arrayshow.jl

```diff
@@ -302,15 +338,63 @@ function show_nd(io::IO, a::AbstractArray, print_matrix, label_slices)
     end
 end
 
+# print_array: main helper functions for _display
+# typeinfo agnostic
+
+# 0-dimensional arrays
+print_array(io::IO, X::AbstractArray{T,0} where T) = isassigned(X) ?
+    show(io, X[]) :
+    print(io, undef_ref_str)
+
+print_array(io::IO, X::AbstractVecOrMat) = print_matrix(io, X)
+
+print_array(io::IO, X::AbstractArray) = show_nd(io, X, print_matrix, true)
+
+# typeinfo aware
+# implements: show(io::IO, ::MIME"text/plain", X::AbstractArray)
```

Comment:
Because basically this method of `show` is what is invoked by `display`, whereas calling `show(object)` prints something else.

Reply:
Call it `showtextplain`, then?   It's not *only* called by `display`, nor is it necessary called by `display` (depends on the `Display`), and confusing `display` overloading with `show` overloading is a perennial problem.

Not sure what was wrong with the old `showarray`, though… can't we leave the name as-is?

Reply:
@rfourquet Bump!

Reply:
IIRC, the old `showarray` was doing mainly two distinct things, which have been separated out into two distinct functions here, so it made sense to use new names: otherwise which one should inherit the old name? But maybe there is an easy answer (which also doesn't break code which may have used this function)? (I can't look at this right now).